lib/src/drawing_and_graphics/shadow.dart

```diff
@@ -0,0 +1,129 @@
+import 'dart:ui';
+
+import 'package:flutter/widgets.dart';
+
+/// A widget that adds a shadow effect to its child widget.
+class Shadow extends StatelessWidget {
+  /// Creates a Shadow widget with the given properties.
+  ///
+  /// Use the [radius] to set the amount of blur, where `0.0` is sharp and a higher
+  /// value is more fuzzy. You can also set the shadow [color] and ([x], [y]) offset.
+  const Shadow({
+    super.key,
+    // In SwiftUI the default is `Color(.sRGBLinear, white: 0, opacity: 0.33)`
+    this.color = const Color(0x54000000),
+    required this.radius,
+    this.x = 0,
+    this.y = 0,
+    required this.child,
+  });
+
+  /// The color of the shadow.
+  final Color color;
+
+  /// The blur radius of the shadow.
+  ///
+  /// `0.0` is sharp while higher values are more fuzzy.
+  final double radius;
+
+  /// The horizontal offset of the shadow.
+  final double x;
+
+  /// The vertical offset of the shadow.
+  final double y;
+
+  /// The widget to which the shadow effect will be applied.
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: [
+        Transform.translate(
+          offset: Offset(x, y),
+          child: Blur(
+            sigmaX: radius,
+            sigmaY: radius,
+            child: Tint(
+              color: color,
+              child: Desaturate(
+                child: child,
+              ),
+            ),
+          ),
+        ),
+        child
+      ],
+    );
+  }
+}
+
+class Blur extends StatelessWidget {
+  final Widget child;
```

Comment:
Properties always go below constructors. The only thing that should appear above constructors in a class are static members.